### PR TITLE
Handle path changes when updating load test preferences

### PR DESCRIPTION
**Pull request by markcor11**
Originated from Issue: https://github.com/eclipse/codewind/issues/164 

# Problem

After changing the PATH field under the load test preferences dialog,  clicking save closes the settings dialog box,  however the chart does will refresh and instead displays the red error handler "Unable to render". 

Screenshot 2019-08-20 at 15 40 00

# Workaround 

Manually refresh the browser page

# Solution

* This PR ensures that when changing the path,  the counter filter is re-applied using the new path value. 
* Since each project type produces path URLs formatted differently,  (Java for example includes protocol and port information, some include query strings etc where as NodeJS does not) the change also normalises the path into its simplest form by stripping off the server/port and query string.

# Additions 

With the chart now showing just HTTP plots for a specific path, these changes ensure non-HTTP counters such as CPU and Memory snapshots are also excluded from the charts if they were measured for an endpoint other than the filtered path. By synchronising the metrics across all types and filtering them by path, the charts behaviour is more predictable. 

Ran existing test bucket : 

Test Suites: 6 passed, 6 total
Tests:       41 passed, 41 total

### CODE REVIEW

File: src/performance/dashboard/src/modules/MetricsUtils.js [hunk 1]
```diff
@@ -218,125 +208,92 @@ let getPathFromURL = function (url) {
  */
 let buildChartDataHTTP = function (params) {
 
-    const mcAPIData = params.projectMetrics;
     const scaleFactor = params.scaleFactor;
```
Comment (rwalle61) [inline src/performance/dashboard/src/modules/MetricsUtils.js]: nit: why not
```suggestion
    const { scaleFactor, counterName, urlFilter, decimals, filteredData } = params;
```
?

File: src/performance/dashboard/src/modules/MetricsUtils.js [hunk 2]
```diff
@@ -218,125 +208,92 @@ let getPathFromURL = function (url) {
...
-            return snapshot["time"]
-        }));
+    const http = metrics.find(metric => { return metric.type === "http" });
+    http.metrics.forEach(snapshot => {
+        const urlList = snapshot.value.data;
+        const foundUrlSamples = urlList.find(urlRow => {
+            if (urlRow && urlRow.url) {
+                let uri = getPathFromURL(urlRow.url);
+                return getEndpoint(uri) === filteredUrl;
+            } return false;
```
Comment (rwalle61) [inline src/performance/dashboard/src/modules/MetricsUtils.js]: nit:
```suggestion
            } 
            return false;
```

File: src/performance/dashboard/src/modules/MetricsUtils.js [hunk 3]
```diff
@@ -218,125 +208,92 @@ let getPathFromURL = function (url) {
...
-
+    // merge all the metrics into the data model using a timestamp key
     metrics.forEach(metricType => {
         metricType.metrics.forEach(snapshot => {
-            let timestamp = model.find(element => { return element['time'] === snapshot['time'] })
-            let metricsGroup = {}
-            metricsGroup.type = metricType.type;
-            metricsGroup.value = snapshot;
-            timestamp[metricType.type] = metricsGroup;
+            const timestamp = model.find(element => { return element['time'] === snapshot['time'] });
```
Comment (rwalle61) [inline src/performance/dashboard/src/modules/MetricsUtils.js]: nit: why not
```suggestion
            const timestamp = model.find(element => { return element.time === snapshot.time });
```

File: src/performance/dashboard/src/modules/MetricsUtils.js (hunk 1 above)
Comment (markcor11) [reply src/performance/dashboard/src/modules/MetricsUtils.js]: > question: why not

that would work too,  likely they didn't always have the same name

**Pull request merged**